store/src/java/com/zimbra/cs/service/formatter/LdifFormatter.java

```diff
@@ -86,77 +105,187 @@ public void formatCallback(UserServletContext context) throws UserServletExcepti
         context.resp.getWriter().print(sb.toString());
     }
 
-    public void toLDIFContact(List<String> fields, Map<String, String> contact, StringBuilder sb) {
-        int NUL = 0;
-        int LF = 10;
-        int CR = 13;
-        int SPACE = 32;
-        int COLON = 58;
-        int LESSTHAN = 60;
-        for (String field : fields) {
-            String value = contact.get(field);
+    public void toLDIFContact(List<String> fields, Map<String, String> contact, StringBuilder sb, String[] galLdapAttrMap) {
+        String email = contact.get(ContactConstants.A_email);
+        if (email != null) {
+            String uid = null;
+            String[] dc = null;
+            List<String> dnElements = new ArrayList<String>();
+            uid = extractUid(email);
+            dc = extractDc(email);
+
+            if (uid != null) {
+                dnElements.add(LdapConstants.ATTR_uid + "=" + uid);
+            }
+            dnElements.add(LdapConstants.ATTR_ou + "=" + LdapConstants.OU_VALUE);
+            if (dc != null) {
+                for (String dcPart : dc) {
+                    dnElements.add(LdapConstants.ATTR_dc + "=" + dcPart);
+                }
+            }
+            String dnValue = StringUtils.join(dnElements.toArray(), ",");
+            constructLDIFEntry(LdapConstants.ATTR_dn, dnValue, sb, false);
+        } else {
+            constructLDIFEntry(LdapConstants.ATTR_dn, "", sb, false);
+        }
+        constructLDIFEntry(LdapConstants.ATTR_objectClass, LdapObjectClass.ZIMBRA_DEFAULT_PERSON_OC, sb, false);
+        for(String attr : galLdapAttrMap) {
+            LdapGalMapRule rule = new LdapGalMapRule(attr, null);
+            addLDIFEntry(rule,sb,contact,ContactConstants.A_fullName, true);
```

Comment:
as discussed, please return a value to indicate field has been mapped and continue to next string.

Reply:
done